Just warn on missing arch

Some architectures are not listed in dpkg list (e.g. mips64el, xtensa, sh4a).
Don't break the build for that.

Signed-off-by: Baruch Siach <baruch@tkos.co.il>
---
diff -Nuar dpkg-1.16.15-orig/m4/dpkg-arch.m4 dpkg-1.16.15/m4/dpkg-arch.m4
--- dpkg-1.16.15-orig/m4/dpkg-arch.m4	2014-05-30 02:47:25.000000000 +0300
+++ dpkg-1.16.15/m4/dpkg-arch.m4	2015-01-16 06:55:19.701596362 +0200
@@ -57,7 +57,7 @@
 AC_MSG_CHECKING([dpkg architecture name])
 _DPKG_ARCHITECTURE([DEB_HOST_ARCH], [dpkg_arch])
 if test "x$dpkg_arch" = "x"; then
-	AC_MSG_ERROR([cannot determine host dpkg architecture])
+	AC_MSG_WARN([cannot determine host dpkg architecture])
 else
 	AC_MSG_RESULT([$dpkg_arch])
 fi
